Add Sysbox as an option to run kubernetes inside unprivileged containers or pods. (#31593)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubelet-in-userns.md

<!-- Supporting rootless podman is discussed in https://github.com/kubernetes/minikube/issues/8719 -->

## Running Kubernetes inside Unprivileged Containers

{{% thirdparty-content %}}

### sysbox

[Sysbox](https://github.com/nestybox/sysbox) is an open-source container runtime
(similar to "runc") that supports running system-level workloads such as Docker
and Kubernetes inside unprivileged containers isolated with the Linux user
namespace.

> See Kubernetes-in-Docker in the [Sysbox docs][https://github.com/nestybox/sysbox/blob/master/docs/quickstart/kind.md] for more info.

Reviewer: ```suggestion
See [Sysbox Quick Start Guide: Kubernetes-in-Docker](https://github.com/nestybox/sysbox/blob/master/docs/quickstart/kind.md) for more information.
```

Author: Done, thanks for the quick review!

> Note that Sysbox supports running Kubernetes inside unprivileged containers

Reviewer: nit (optional fix)
```suggestion
Sysbox supports running Kubernetes inside unprivileged containers
```

Author: Done.

without requiring Cgroup v2 and without the `KubeletInUserNamespace` feature

> gate. It does this by exposing specially crafted proc and sys filesystems inside

Reviewer: nit (optional fix)
```suggestion
gate. It does this by exposing specially crafted `/proc` and `/sys` filesystems inside
```

Author: Done.
